spec/controllers/recordings_controller_spec.rb

```diff
@@ -27,6 +27,24 @@
       response_recording_ids = JSON.parse(response.body)['data'].map { |recording| recording['id'] }
       expect(response_recording_ids).to be_empty
     end
+
+    it 'returns the proper recordings according to the query' do
+      course_name = Faker::Educator.course_name
+      course_recordings = []
+
+      recordings = create_list(:recording, 10) do |recording, i|
+        if i.even?
+          recording.name = course_name
+          course_recordings << recording
+        end
+      end
+
+      create(:room, user:, recordings:)
+
+      get :index, params: { search: course_name }
+      response_recording_ids = JSON.parse(response.body)['data'].map { |recording| recording['id'] }
+      expect(response_recording_ids).to eq(course_recordings.pluck(:id))
+    end
```

Comment:
I would take this a step further - manually create some recordings with some set names and then manually set the search param to a string and make sure the recordings returned are correct